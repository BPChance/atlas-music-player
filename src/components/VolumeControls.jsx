import volume from "../assets/VolumeIcon.svg";

export default function VolumeControls() {
  return (
    <div className="flex items-center">
      <span>
        <img src={volume} alt="volume" />
      </span>
      <input
        type="range"
        min="0"
        max="100"
        defaultValue="50"
        className="h-[15px] w-[374px] cursor-pointer bg-[#D5D7D8] accent-[#64748B]"
      />
    </div>
  );
}
