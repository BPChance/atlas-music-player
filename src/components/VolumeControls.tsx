import volume from "../assets/VolumeIcon.svg";

export default function VolumeControls() {
  return (
    <div className="flex w-full items-center gap-2">
      <span>
        <img src={volume} alt="volume" />
      </span>
      <input
        type="range"
        min="0"
        max="100"
        defaultValue="50"
        className="accent-supapink h-[15px] w-full max-w-[374px] cursor-pointer bg-[#D5D7D8] shadow-2xl"
      />
    </div>
  );
}
