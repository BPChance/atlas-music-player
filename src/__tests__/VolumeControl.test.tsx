import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import VolumeControls from "../components/VolumeControls";

describe("VolumeControls", () => {
  it("renders full volume", () => {
    const { container } = render(
      <VolumeControls volume={1} setVolume={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders half volume", () => {
    const { container } = render(
      <VolumeControls volume={0.5} setVolume={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders mute volume", () => {
    const { container } = render(
      <VolumeControls volume={0} setVolume={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
});
