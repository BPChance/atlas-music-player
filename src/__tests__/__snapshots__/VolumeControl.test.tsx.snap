// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls > renders full volume 1`] = `
<div>
  <div
    class="flex w-full items-center gap-2"
  >
    <span>
      <img
        alt="volume"
        src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M11%204.70203C10.9998%204.56274%2010.9583%204.42663%2010.8809%204.31088C10.8034%204.19514%2010.6934%204.10493%2010.5647%204.05166C10.436%203.99838%2010.2944%203.98442%2010.1577%204.01154C10.0211%204.03866%209.89559%204.10564%209.797%204.20403L6.413%207.58703C6.2824%207.7184%206.12703%207.82256%205.95589%207.89345C5.78475%207.96435%205.60124%208.00057%205.416%208.00003H3C2.73478%208.00003%202.48043%208.10539%202.29289%208.29292C2.10536%208.48046%202%208.73481%202%209.00003V15C2%2015.2652%202.10536%2015.5196%202.29289%2015.7071C2.48043%2015.8947%202.73478%2016%203%2016H5.416C5.60124%2015.9995%205.78475%2016.0357%205.95589%2016.1066C6.12703%2016.1775%206.2824%2016.2817%206.413%2016.413L9.796%2019.797C9.8946%2019.8958%2010.0203%2019.9631%2010.1572%2019.9904C10.2941%2020.0177%2010.436%2020.0037%2010.5649%2019.9503C10.6939%2019.8968%2010.804%2019.8063%2010.8815%2019.6902C10.959%2019.5741%2011.0002%2019.4376%2011%2019.298V4.70203Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20fill='black'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20fill='black'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
      />
    </span>
    <input
      class="accent-supapink h-[15px] w-full max-w-[374px] cursor-pointer bg-[#D5D7D8] shadow-2xl"
      max="1"
      min="0"
      step="0.01"
      type="range"
      value="1"
    />
  </div>
</div>
`;

exports[`VolumeControls > renders half volume 1`] = `
<div>
  <div
    class="flex w-full items-center gap-2"
  >
    <span>
      <img
        alt="volume"
        src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M11%204.70203C10.9998%204.56274%2010.9583%204.42663%2010.8809%204.31088C10.8034%204.19514%2010.6934%204.10493%2010.5647%204.05166C10.436%203.99838%2010.2944%203.98442%2010.1577%204.01154C10.0211%204.03866%209.89559%204.10564%209.797%204.20403L6.413%207.58703C6.2824%207.7184%206.12703%207.82256%205.95589%207.89345C5.78475%207.96435%205.60124%208.00057%205.416%208.00003H3C2.73478%208.00003%202.48043%208.10539%202.29289%208.29292C2.10536%208.48046%202%208.73481%202%209.00003V15C2%2015.2652%202.10536%2015.5196%202.29289%2015.7071C2.48043%2015.8947%202.73478%2016%203%2016H5.416C5.60124%2015.9995%205.78475%2016.0357%205.95589%2016.1066C6.12703%2016.1775%206.2824%2016.2817%206.413%2016.413L9.796%2019.797C9.8946%2019.8958%2010.0203%2019.9631%2010.1572%2019.9904C10.2941%2020.0177%2010.436%2020.0037%2010.5649%2019.9503C10.6939%2019.8968%2010.804%2019.8063%2010.8815%2019.6902C10.959%2019.5741%2011.0002%2019.4376%2011%2019.298V4.70203Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20fill='black'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20fill='black'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
      />
    </span>
    <input
      class="accent-supapink h-[15px] w-full max-w-[374px] cursor-pointer bg-[#D5D7D8] shadow-2xl"
      max="1"
      min="0"
      step="0.01"
      type="range"
      value="0.5"
    />
  </div>
</div>
`;

exports[`VolumeControls > renders mute volume 1`] = `
<div>
  <div
    class="flex w-full items-center gap-2"
  >
    <span>
      <img
        alt="volume"
        src="data:image/svg+xml,%3csvg%20width='24'%20height='24'%20viewBox='0%200%2024%2024'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M11%204.70203C10.9998%204.56274%2010.9583%204.42663%2010.8809%204.31088C10.8034%204.19514%2010.6934%204.10493%2010.5647%204.05166C10.436%203.99838%2010.2944%203.98442%2010.1577%204.01154C10.0211%204.03866%209.89559%204.10564%209.797%204.20403L6.413%207.58703C6.2824%207.7184%206.12703%207.82256%205.95589%207.89345C5.78475%207.96435%205.60124%208.00057%205.416%208.00003H3C2.73478%208.00003%202.48043%208.10539%202.29289%208.29292C2.10536%208.48046%202%208.73481%202%209.00003V15C2%2015.2652%202.10536%2015.5196%202.29289%2015.7071C2.48043%2015.8947%202.73478%2016%203%2016H5.416C5.60124%2015.9995%205.78475%2016.0357%205.95589%2016.1066C6.12703%2016.1775%206.2824%2016.2817%206.413%2016.413L9.796%2019.797C9.8946%2019.8958%2010.0203%2019.9631%2010.1572%2019.9904C10.2941%2020.0177%2010.436%2020.0037%2010.5649%2019.9503C10.6939%2019.8968%2010.804%2019.8063%2010.8815%2019.6902C10.959%2019.5741%2011.0002%2019.4376%2011%2019.298V4.70203Z'%20fill='black'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20fill='black'/%3e%3cpath%20d='M16%209C16.6491%209.86548%2017%2010.9181%2017%2012C17%2013.0819%2016.6491%2014.1345%2016%2015'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20fill='black'/%3e%3cpath%20d='M19.364%2018.364C20.1998%2017.5283%2020.8627%2016.5361%2021.315%2015.4442C21.7673%2014.3522%2022.0001%2013.1819%2022.0001%2012C22.0001%2010.8181%2021.7673%209.64775%2021.315%208.55581C20.8627%207.46387%2020.1998%206.47172%2019.364%205.63599'%20stroke='black'%20stroke-width='2'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3c/svg%3e"
      />
    </span>
    <input
      class="accent-supapink h-[15px] w-full max-w-[374px] cursor-pointer bg-[#D5D7D8] shadow-2xl"
      max="1"
      min="0"
      step="0.01"
      type="range"
      value="0"
    />
  </div>
</div>
`;
